packages/govuk-frontend/src/govuk/components/accordion/accordion.mjs

```diff
@@ -235,19 +234,19 @@ export class Accordion extends GOVUKFrontendComponent {
       return
     }
 
-    // Create a button element that will replace the '.govuk-accordion__section-button' span
+    // Create a button element that will replace the
+    // '.govuk-accordion__section-button' span
     const $button = document.createElement('button')
     $button.setAttribute('type', 'button')
     $button.setAttribute(
       'aria-controls',
       `${this.$module.id}-content-${index + 1}`
     )
 
-    // Copy all attributes (https://developer.mozilla.org/en-US/docs/Web/API/Element/attributes) from $span to $button
+    // Copy all attributes from $span to $button (except `id`, which gets added
+    // to the `$headingText` element)
     for (let i = 0; i < $span.attributes.length; i++) {
       const attr = $span.attributes.item(i)
-      // Add all attributes but not ID as this is being added to
```

Comment:
*nitpick* Wondering if that comment is not actually useful because it explains that ID will get its own special treatment with the heading text, so may help not miss the `$headingText.id = $span.id`.

Reply:
I've moved it to the comment before the loop – think that's too far away from the code?

Reply:
Oh, I'd missed it, sorry. Yeah, all good 🥳